irohad/simulator/impl/simulator.cpp

```diff
@@ -149,6 +149,8 @@ namespace iroha {
                                             rejected_hashes);
       crypto_signer_->sign(*block);
 
+      last_block.reset();
```

Comment:
please add a TODO to refactor `processVerifiedProposal` in such a way that it accepts both verified proposal and last block, and is called from `processProposal` which provides it the last block as an argument. This will let us remove the ugly `last_block` which is now sometimes nulled and never tested for null...